src/utils/sorted-electron-map.ts

```diff
@@ -20,12 +20,27 @@ export function sortedElectronMap<T>(
         return a.localeCompare(b);
       }
 
-      if (!semver.valid(a)) {
-        return -1;
-      }
+      if (!semver.valid(a)) return -1;
+      if (!semver.valid(b)) return 1;
+
+      // Handle prerelease tag sorting - semver does not handle this.
+      // See https://www.npmjs.com/package/semver#prerelease-tags for more.
+      if (semver.prerelease(a) && semver.prerelease(b)) {
+        const [tagA, valueA] = semver.prerelease(a)!;
+        const [tagB, valueB] = semver.prerelease(b)!;
 
-      if (!semver.valid(b)) {
-        return 1;
+        if (semver.major(a) === semver.major(b)) {
+          if (tagA === 'beta' && tagA === tagB) {
+            // Both are betas - compare beta number.
+            return valueA > valueB ? -1 : 1;
+          } else if (tagA === 'nightly' && tagA === tagB) {
+            // Both are nightlies - compare nightly date.
+            return new Date(valueA) > new Date(valueB) ? -1 : 1;
```

Comment:
no need to convert these into dates as the strings themselves will sort the right way.
```suggestion
            return valueA.localeCompare(valueB);
```